packages/@aws-cdk/aws-codepipeline-api/lib/action.ts

```diff
@@ -141,6 +141,15 @@ export interface CommonActionProps {
    * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/reference-pipeline-structure.html
    */
   runOrder?: number;
+
+  /**
+   * The service role that is assumed during execution of action.
+   * This role is not mandatory, however more advanced configuration
+   * may require specifying it.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-codepipeline-pipeline-stages-actions.html
+   */
+  actionRole?: iam.IRole;
```

Comment:
This is a good start :) but is not enough to be passed from every role back to the `Role` class. You have to make sure the constructors of every concrete Action class call `super` with the correct parameters. The best way is to do it is to use the [spread operator](https://basarat.gitbooks.io/typescript/docs/spread-operator.html), like this:

```ts
class SomeAction extends AnotherAction {
  constructor(parent: cdk.Construct, id: string, props: SomeActionProps) {
    super(parent, id, {
      // some properties passed explicitly
      ...props,
    });
  }
}
```

This way, you're guaranteed all of the properties will be automatically passed (the reason some of them don't use it yet is because it was written for an older TypeScript version).

Reply:
I added passing the action's role for other actions. I wonder if we would like to refactor existing code to use spread operator, so in future adding new attributes to action's interfaces would be more transparent?

Reply:
Yes, every Action subclass that does not currently use the spread operator should be changed to use it. I thought I made it clear in my previous comment, but I guess I failed :(

Reply:
Replaced
```
stage: props.stage,
runOrder: props.runOrder
```
with
```
...props
```